src/test/java/com/crud/tasks/controller/TaskControllerTestSuite.java

```diff
@@ -0,0 +1,111 @@
+
+import com.crud.tasks.domain.TaskDto;
+import com.google.gson.Gson;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(TaskController.class)
+public class TaskControllerTestSuite {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private TaskController taskController;
+
+    @Test
+    public void shouldFetchTasks() throws Exception {
+        //Given
+        List<TaskDto> taskDtoList = new ArrayList<>();
+        taskDtoList.add(new TaskDto(1L, "Test List", "Test Content"));
+        Mockito.when(taskController.getTasks()).thenReturn(taskDtoList);
+        //When & Then
+        mockMvc.perform(get("/v1/task/getTasks").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", Matchers.hasSize(1)))
+                .andExpect(jsonPath("$[0].id", Matchers.is(1)))
+                .andExpect(jsonPath("$[0].title", Matchers.is("Test List")))
+                .andExpect(jsonPath("$[0].content", Matchers.is("Test Content")));
+    }
+
+    @Test
+    public void shouldFetchTask() throws Exception {
+        //Given
+        TaskDto taskDto = new TaskDto(1L, "Test Task", "Test Content");
+        Mockito.when(taskController.getTask(taskDto.getId())).thenReturn(taskDto);
+        //When & Then
+        mockMvc.perform(get("/v1/task/getTask")
+                .contentType(MediaType.APPLICATION_JSON)
+                .param("taskId", "1"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", Matchers.is(1)))
+                .andExpect(jsonPath("$.title", Matchers.is("Test Task")))
+                .andExpect(jsonPath("$.content", Matchers.is("Test Content")));
+    }
+
+    @Test
+    public void shouldDeleteTask() throws Exception {
+        //Given
+        Mockito.doNothing().when(taskController).deleteTask(1L);
+        //When & Then
+        mockMvc.perform(delete("/v1/task/deleteTask")
+                .contentType(MediaType.APPLICATION_JSON)
+                .param("taskId", "1"))
+                .andExpect(status().isOk());
+        Mockito.verify(taskController, Mockito.times(1)).deleteTask(1L);
```

Comment:
w sumie tego verify nie potrzebujesz, bo sprawdza to perform

Reply:
ok, poprawie to w kolejnym commit